components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -7910,7 +7910,13 @@ public API getAPIbyUUID(String uuid, String organization) throws APIManagementEx
                 throw new APIMgtResourceNotFoundException(msg);
             }
         } catch (APIPersistenceException e) {
-            throw new APIManagementException("Failed to get API", e);
+            if (e.getErrorHandler().getErrorCode() == ExceptionCodes.INTERNAL_ERROR.getErrorCode()){
```

Comment:
What are the error handler properties during this case?
If we filter INTERNAL_ERRORs here and send a 404, that will be sent for other real errors as well.

Problem is at this point where we are sending a general APIPersistenceException without specifying an error code.
https://github.com/wso2-enterprise/choreo-apimgt-extensions/blob/3986beace8154c3d130b4252c7979f7261ff66f5/components/org.wso2.am.choreo.extensions.persistence.mongodb/src/main/java/org/wso2/am/choreo/extensions/persistence/mongodb/MongoDBPersistenceSingleCollectionImpl.java#L330-L333

Reply:
These are the error handler properties during this case.
![Screenshot from 2022-08-25 10-57-02](https://user-images.githubusercontent.com/35162901/186582210-cf81b659-1e7f-4417-9efd-94969fa5fa52.png)

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/11395/commits/33f8b4d9f127da243076390d4113f01e974659ac by handling the specific exception from the persistence layer